### PR TITLE
vlan_bridging_qinq: clarify state of stacked VLAN support

### PR DESCRIPTION
**Pull request by KanjiMonster**
Stacked VLAN support is not really supported and implemented, and mostly works due to defaults. Since there may be some expectations, document the current behavior and its limitations.

### CODE REVIEW

File: network_configuration/vlan_bridging_qinq.md [hunk 1]
```diff
@@ -14,6 +14,9 @@ Similarly to 802.1q bridging, it is possible to configure 802.1ad VLANs using [i
 **WARNING**: Any bridge configured to forward VLAN traffic with either protocol 802.1q or 802.1ad will only forward traffic of the selected VLAN protocol type.
 {: .label .label-red }
 
+**WARNING**: Stacking VLANs is currently mostly unsupported. See [stacked VLAN support](#stacked-vlan-support) for current behavior and limitations.
```
Comment (jklare) [inline network_configuration/vlan_bridging_qinq.md:17]:
```suggestion
**WARNING**: Stacking VLANs and switching based on both tags is currently not supported. See [stacked VLAN support](#stacked-vlan-support) for current behavior and limitations.
```

File: network_configuration/vlan_bridging_qinq.md [hunk 2]
```diff
@@ -45,3 +48,13 @@ VLANProtocol=802.1ad
 ```
 
 The remaining configurations follow the same steps from the instructions in the [802.1q bridging](/network_configuration/vlan_bridging.html#systemd-networkd) section.
+
+## stacked VLAN support
+
+Support for stacking VLANs on the Linux side is currently unimplemented. The switch will forward double tagged packets according to its S-TAG, as long as the S-TAG matches the configured VLAN protocol of the bridge.
```
Comment (jklare) [inline network_configuration/vlan_bridging_qinq.md:54]:
```suggestion
Support for stacking VLANs is currently not implemented on the Linux side. The switch will forward double tagged packets according to its S-TAG, as long as the S-TAG matches the configured VLAN protocol of the bridge.
```

File: network_configuration/vlan_bridging_qinq.md [hunk 3]
```diff
@@ -45,3 +48,13 @@ VLANProtocol=802.1ad
...
+
+The following limitations apply:
+
+* Learning is done based on the S-TAG, so all inner VLANs share one learning database.
+* PVID port configuration only works for untagged packets. Single tagged packets with an unexpected VLAN protocol will not be converted to double tagged and instead dropped.
```
Comment (jklare) [inline network_configuration/vlan_bridging_qinq.md:59]:
```suggestion
* PVID port configuration only works for untagged packets. Single tagged packets with an unexpected VLAN protocol will not be converted to double tagged and be dropped instead.
```

File: network_configuration/vlan_bridging_qinq.md [hunk 4]
```diff
@@ -45,3 +48,13 @@ VLANProtocol=802.1ad
...
+* Adding a VLAN interface to the bridge is not supported.
```
Comment (jklare) [inline network_configuration/vlan_bridging_qinq.md:60]: This seems rather unspecific. Maybe you could add a couple of details for which case this is true.

File: network_configuration/vlan_bridging_qinq.md (hunk 1 above)
Comment (KanjiMonster) [reply network_configuration/vlan_bridging_qinq.md]: Added.

File: network_configuration/vlan_bridging_qinq.md (hunk 3 above)
Comment (KanjiMonster) [reply network_configuration/vlan_bridging_qinq.md]: Added.

File: network_configuration/vlan_bridging_qinq.md (hunk 2 above)
Comment (KanjiMonster) [reply network_configuration/vlan_bridging_qinq.md]: Added.

File: network_configuration/vlan_bridging_qinq.md (hunk 4 above)
Comment (KanjiMonster) [reply network_configuration/vlan_bridging_qinq.md]: Reworded and expanded it a bit.

File: network_configuration/vlan_bridging_qinq.md [hunk 5]
```diff
@@ -45,3 +48,13 @@ VLANProtocol=802.1ad
 ```
 
 The remaining configurations follow the same steps from the instructions in the [802.1q bridging](/network_configuration/vlan_bridging.html#systemd-networkd) section.
+
+## stacked VLAN support
+
+Support for configuring stacked VLANs in any way is currently not implemented. The switch will still forward double tagged packets according to its outer VLAN tag, as long as the outer VLAN tag's protoctol matches the configured VLAN protocol of the bridge.
```
Comment (jklare) [inline network_configuration/vlan_bridging_qinq.md:54]:
```suggestion
Support for configuring stacked VLANs in any way is currently not implemented. The switch will still forward double tagged packets according to their outer VLAN tag, as long as the outer VLAN tag's protoctol matches the configured VLAN protocol of the bridge.
```

Comment (KanjiMonster) [reply network_configuration/vlan_bridging_qinq.md]: Fixed.

**Pull request merged**